components/search/index.css

```diff
@@ -10,21 +10,173 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+    --highcontrast-search-color-hover: CanvasText;
+    --highcontrast-search-color-focus: CanvasText;
+    --highcontrast-search-color-disabled: GrayText;
+  }
+
+  .spectrum-Search .spectrum-Search-clearButton {
+    /* Offset from edges to avoid ButtonFace background color overlap on border. */
+    inset-inline-end: calc(var(--mod-search-border-width, var(--spectrum-search-border-width)) + 1px);
+    inset-block-start: calc(var(--mod-search-border-width, var(--spectrum-search-border-width)) + 1px);
+  
+    .spectrum-ClearButton-fill {
+      height: calc(var(--mod-search-block-size, var(--spectrum-search-block-size)) - (2 * var(--mod-search-border-width, var(--spectrum-search-border-width))) - 2px);
+      width: calc(var(--mod-search-block-size, var(--spectrum-search-block-size)) - (2 * var(--mod-search-border-width, var(--spectrum-search-border-width))) - 2px);
+      border-radius: calc(var(--spectrum-search-border-radius) - var(--mod-search-border-width, var(--spectrum-search-border-width)) - 1px);
+    }
+  }
 }
 
 .spectrum-Search {
   display: inline-block;
   position: relative;
+  inline-size: var(--mod-search-inline-size, var(--spectrum-search-inline-size));
+  min-inline-size: var(--mod-search-min-inline-size, var(--spectrum-search-min-inline-size));
+
+  .spectrum-HelpText {
+    margin-block-start: var(--mod-search-to-help-text, var(--spectrum-search-to-help-text));
+  }
 }
 
 .spectrum-Search-clearButton {
   position: absolute;
   inset-inline-end: 0;
   inset-block-start: 0;
+  border-radius: var(--mod-search-border-radius, var(--spectrum-search-border-radius));
+
+  .spectrum-ClearButton-fill {
+    border-radius: var(--mod-search-border-radius, var(--spectrum-search-border-radius));
+  }
+
+  .spectrum-Search.is-disabled & {
+    display: none;
+  }
+}
+
+.spectrum-Search-textfield {
+  inline-size: 100%;
+}
+
+.spectrum-Search-icon {
+  color: var(--highcontrast-search-color-default, var(--mod-search-color-default, var(--spectrum-search-color-default)));
+
+  .spectrum-Search-textfield:hover & {
+    color: var(--highcontrast-search-color-hover, var(--mod-search-color-hover, var(--spectrum-search-color-hover)));
+  }
+
+  .spectrum-Search-textfield.is-focused & {
+    color: var(--highcontrast-search-color-focus, var(--mod-search-color-focus, var(--spectrum-search-color-focus)));
+  }
+
+  .spectrum-Search-textfield.is-focused:hover & {
+    color: var(--highcontrast-search-color-focus, var(--mod-search-color-focus-hover, var(--spectrum-search-color-focus-hover)));
+  }
+
+  .spectrum-Search-textfield.is-keyboardFocused & {
+    color: var(--highcontrast-search-color-focus, var(--mod-search-color-key-focus, var(--spectrum-search-color-key-focus)));
+  }
+
+  .spectrum-Search-textfield.is-disabled &,
+  .spectrum-Search-textfield.is-disabled:hover & {
+    color: var(--highcontrast-search-color-disabled, var(--mod-search-color-disabled, var(--spectrum-search-color-disabled)));
+  }
 }
```

Comment:
```suggestion
.spectrum-Search-icon {
  --spectrum-search-color: var(--highcontrast-search-color-default, var(--mod-search-color-default, var(--spectrum-search-color-default)));
  color: var(--spectrum-search-color);

  .spectrum-Search-textfield:hover & {
    --spectrum-search-color: var(--highcontrast-search-color-hover, var(--mod-search-color-hover, var(--spectrum-search-color-hover)));
  }

  .spectrum-Search-textfield.is-focused & {
    --spectrum-search-color: var(--highcontrast-search-color-focus, var(--mod-search-color-focus, var(--spectrum-search-color-focus)));
  }

  .spectrum-Search-textfield.is-focused:hover & {
    --spectrum-search-color: var(--highcontrast-search-color-focus, var(--mod-search-color-focus-hover, var(--spectrum-search-color-focus-hover)));
  }

  .spectrum-Search-textfield.is-keyboardFocused & {
    --spectrum-search-color: var(--highcontrast-search-color-focus, var(--mod-search-color-key-focus, var(--spectrum-search-color-key-focus)));
  }

  .spectrum-Search-textfield.is-disabled &,
  .spectrum-Search-textfield.is-disabled:hover & {
    --spectrum-search-color: var(--highcontrast-search-color-disabled, var(--mod-search-color-disabled, var(--spectrum-search-color-disabled)));
  }
}
```

This seems like a great place to use a local variable approach with variable overrides based on the specificity of the selectors. What do you think?

Reply:
There's a couple of benefits to updating custom properties instead of css properties for state:

- reusability: you can use the same updated variable to apply to multiple properties (i.e., calculations) or influence multiple elements (i.e., children element changes) but can update the value without having to redefine every place it was used.
- clarity: since the values cascade, we don't have to define them on the exact element they apply to, we can instead apply all state-based changes directly to the state's class and let those values influence any children they might need to.

Reply:
I've made your suggested change in the latest series of commits.